litgpt/lora.py

```diff
@@ -159,6 +156,10 @@ def merge(self) -> None:
                 # assign updated weights and quantize by moving to CUDA device
                 self.linear.weight = bnb.nn.Params4bit(weight_data, requires_grad=False, **weight.__dict__)
                 self.linear.weight.cuda(weight.device)
+            # if the pretrained weights and LoRA weights are of compatible dtypes - simply sum them
+            elif torch.finfo(pretrained_dtype).max >= torch.finfo(lora_data.dtype).max:
```

Comment:
It's a nice guard to have, but could you name an example when such a mismatch in dtype can be a case?

Reply:
In the previous code, if pretrained weights are in float32 and lora weights are in float16, the `NotImplementedError` gets raised. This happens for instance with the stablelm 3b checkpoint.

My intention with this change is to allow the merge if the target dtype has a larger or equal range

Reply:
Now I get it. Forgot about my own `else` case 😆 
Thanks